cookbooks/bcpc-hadoop/libraries/hive_metastore.rb

```diff
@@ -0,0 +1,96 @@
+require 'poise'
+
+module BcpcHadoop
+  module HiveMetastore
+    module Database
+    end
+  end
+end
+
+module BcpcHadoop::HiveMetastore::Database
+  class UpgradeError < RuntimeError; end
+
+  class Resource < Chef::Resource
+    include Poise
+
+    provides :hive_metastore_database
+
+    attribute :root_password, kind_of: String
+    attribute :hive_password, kind_of: String
+    attribute :schematool_path, kind_of: String
+
+    actions :create, :init, :upgrade
+  end
+
+  class Provider < Chef::Provider
+    include Poise
+
+    provides :hive_metastore_database
+
+    def action_create
+      return if database_created?
+      converge_by 'creating the metastore database' do
+        create_hive_database
+      end
+    end
+
+    def create_hive_database
+      privs = 'SELECT,INSERT,UPDATE,DELETE,LOCK TABLES,EXECUTE'
+      command = Mixlib::ShellOut.new "mysql -uroot -p#{new_resource.root_password}",
+        input: <<-eos
+CREATE DATABASE metastore;
```

Comment:
In addition, isn't there some MySQL provider already? I'll check the code and update this comment if I find one, since it would probably save you lots of work :)